stdlib/uchar.ml

```diff
@@ -20,41 +20,46 @@ let err_no_succ = "U+10FFFF has no successor"
 let err_not_sv i = format_int "%X" i ^ " is not a Unicode scalar value"
 let err_not_latin1 u = "U+" ^ format_int "%04X" u ^ " is not a latin1 character"
 
-type t = int
+type t = uchar
+external unsafe_of_int : int -> t = "%identity"
+external to_int : t -> int = "%identity"
 
-let min = 0x0000
-let max = 0x10FFFF
-let lo_bound = 0xD7FF
-let hi_bound = 0xE000
+let min = unsafe_of_int 0x0000
+let max = unsafe_of_int 0x10FFFF
```

Comment:
Is it not possible to do:

```suggestion
let max = '\u{10FFFF}'
```

and skip the `unsafe_of_int` entirely? Bootstrapping issue maybe?

Reply:
This avoids relying on the interpretation of uchar literals stored in the bootstrapped compiler.